### PR TITLE
[luci] Add create_const helper to QuantizedModelVerifier test

### PR DESCRIPTION
**Pull request by dayo09**
This commit adds create_const helper function to create const node with value

ONE-DCO-1.0-Signed-off-by: Dayoung Lee <dayg502@gmail.com>

-----

From #6618

### CODE REVIEW

File: compiler/luci/pass/src/QuantizedModelVerifier.test.cpp [hunk 1]
```diff
@@ -851,6 +877,29 @@ TEST(QuantizedModelVerifierTest, LocalCreateDummyConst)
...
+}
+
+TEST(QuantizedModelVerifierTest, LocalCreateConst_NEG)
+{
+  loco::Graph g;
+  std::initializer_list<float> values = {0.1, 0, -5, 100};
+  luci::CircleConst *node = create_const<Type::FLOAT32, float>(&g, {2, 2}, values);
+
+  // access out of bound
+  EXPECT_DEATH(node->at<Type::FLOAT32>(4), "Assertion `n < size<DT>()");
```
Comment (dayo09) [inline compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:900]: I guess this don't work with release test, because assert doens't work.

**Comment by dayo09** [comment]
@nnfw-bot test nncc-debug
@nnfw-bot test nncc-release
CI Network Error..

**Comment by seanshpark** [comment]
@nnfw-bot test nncc-debug
@nnfw-bot test nncc-release

**Comment by dayo09** [comment]
### EXPECT_DEATH regex

EXPECT_DEATH(statement, regex) gets regex parameter and expects the statement to be death by a message of which contains regex.
It seems that `'` is regarded as a special character. (but even with escape character `\` doesn't work somehow.)


```
TEST(QuantizedModelVerifierTest, LocalCreateConst_NEG)
{
  ...
  EXPECT_DEATH(node->at<Type::FLOAT32>(4), "Assertion `n < size<DT>()' failed."); //FAIL
}
TEST(QuantizedModelVerifierTest, LocalCreateConst_NEG2)
{
  ...
  EXPECT_DEATH(node->at<Type::FLOAT32>(4), "Assertion `n < size<DT>()"); // SUCCESS
}
```

Full target error message:
```
[  DEATH   ] luci_pass_test: /nfs/git/ONE/compiler/luci/lang/src/Nodes/CircleConst.cpp:48: typename loco::DataTypeImpl<DT>::Type& luci::CircleConst::at(uint32_t) [with loco::DataType DT = (loco::DataType)10; typename loco::DataTypeImpl<DT>::Type = float; uint32_t = unsigned int]: Assertion `n < size<DT>()' failed.
```

File: compiler/luci/pass/src/QuantizedModelVerifier.test.cpp [hunk 2]
```diff
@@ -851,6 +877,29 @@ TEST(QuantizedModelVerifierTest, LocalCreateDummyConst)
   EXPECT_NO_THROW(create_dummy_const<Type::FLOAT32>(&g, {32, 32}));
 }
 
+TEST(QuantizedModelVerifierTest, LocalCreateConst)
```
Comment (seanshpark) [inline compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:880]: `QuantizedModelVerifierTest` Test is used for IR(Op) tests. to distinguish this, how about revise?
- use anoter name, maybe `QuantizedModelVerifierHelperTest`
- use `local_create_const`

Comment (dayo09) [reply compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:880]: > use anoter name, maybe QuantizedModelVerifierHelperTest

I like this idea :-D

> use local_create_const

It's using a seperate namespace in order to bound the scope. Does it still need `local_` prefix?

Comment (seanshpark) [reply compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:880]: > Does it still need local_ prefix?

I don't think so. :)

File: compiler/luci/pass/src/QuantizedModelVerifier.test.cpp [hunk 3]
```diff
@@ -70,6 +70,32 @@ template <Type T> luci::CircleConst *create_dummy_const(loco::Graph *g, luci::te
   return node;
 }
 
+/**
+ * @brief A helper function to create const node with value
+ *
```
Comment (seanshpark) [inline compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:75]: maybe remove this empty line?

File: compiler/luci/pass/src/QuantizedModelVerifier.test.cpp (hunk 1 above)
Comment (seanshpark) [reply compiler/luci/pass/src/QuantizedModelVerifier.test.cpp:900]: yes, assert check is a bit difficult with gtest for this reason.

**Review by seanshpark — APPROVED** [review]
LGTM

**Comment by mhs4670go** [comment]
@dayo09 FYI, c++ compiler have specific [escape sequences rule](https://en.cppreference.com/w/cpp/language/escape). With that, you can make a proper regex that works well in c++.

In short, here's what we should do.
1. `()`(group capturing), `'`, `.`(Matches any character) should be escaped 
```
"Assertion `n < size<DT>\\(\\)\' failed\\."
```
Just `\(` can't be applied because there is no such escape sequence in c++. So, `\\(` can be transformed into `\(`, which is our goal. And so can `\\.`. 

2. > At any rate, I am removing that EXPECT_DEATH checker as it needs more lines to be working at release build check.

AFAIK, there's `EXPECT_DEBUG_DEATH`. It won't need more lines I think.

**Comment by seanshpark** [comment]
`EXPECT_DEBUG_DEATH` is good for testing Debug but this will make difference in release vs debug.
When we had this kind of problem last year, I recall as it will be not good to make differences for testing with build type.

**Comment by dayo09** [comment]
@mhs4670go Thanks for the information! :-D

**Pull request merged**